src/erlazure_insights.erl

```diff
@@ -72,17 +72,28 @@ request_api(Url, Config) ->
 http_request(SubscriptionId, Filter, Select, Config) ->
     URL = "https://management.azure.com/subscriptions/" ++
     SubscriptionId ++
-    "/providers/microsoft.insights/eventtypes/management/values?" ++ "api-version=2014-04-01" ++ 
-    "&%24filter=" ++ Filter ++
-    "&%24select=" ++ Select,
+    "/providers/microsoft.insights/eventtypes/management/values?" ++
+    make_query_string([
+            {"api-version", "2014-04-01"},
+            {"$filter", Filter},
+            {"$select", Select}
+        ]),
     request_api(URL, Config).
 
+make_query_string(Params) ->
+    lists:foldl(fun({Key, Value}, Acc) -> 
+                case Value of
+                    [] -> Acc;
+                    _ -> Acc ++ "&" ++ Key ++ "=" ++ http_uri:encode(Value)
```

Comment:
@lilrooness also I already have similar (but not exactly the same) function there: https://github.com/alertlogic/erlazure/blob/master/src/erlazure_ad_auth.erl#L91-L93

Since we need that functionality non it just one pleace, can you please also do some refactoring, to have just one `urlencode` in `erlazure_http` module? 
also note that `urlencode` should not filter out empty V, since it's not common case, it makes sense only for your particular module, so please do this filtering in the `erlazure_insights`